### PR TITLE
Add endpoint with implemented s3 upload link

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1307

#### Link to staging deployment URL 
https://ui-martinfosco-ui481-api216.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/ui/pull/481

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/api/route-services/samples.js [hunk 1]
```diff
@@ -201,6 +176,50 @@ class SamplesService {
...
+
+    const params = {
+      Bucket: this.sampleFilesBucketName,
+      Key: `${projectUuid}/${sampleUuid}/${fileName}`,
+      Expires: 360,
+    };
+
+    if (cellrangerVersion) {
+      params.Metadata = {
+        cellranger_version: cellrangerVersion,
```
Comment (aerlaut) [inline src/api/route-services/samples.js:192]: Is this a new metadata or do we have this already?

File: src/specs/api.yaml [hunk 2]
```diff
@@ -992,6 +992,26 @@ paths:
           application/json:
             schema:
               $ref: ./models/samples-bodies/SamplesRemoveBody.v1.yaml
+  '/projects/{projectUuid}/samples/{sampleUuid}/{fileName}/uploadLink':
```
Comment (aerlaut) [inline src/specs/api.yaml]: maybe `/uploadUrl` would be more appropriate?

File: src/specs/api.yaml [hunk 3]
```diff
@@ -992,6 +992,26 @@ paths:
...
+    get:
+      summary: Get file upload link
+      description: Get temporary s3 upload link for fileName
+      operationId: uploadSampleFileLink
+      x-eov-operation-id: samples#uploadSampleFileLink
```
Comment (aerlaut) [inline src/specs/api.yaml]: Likewise, I think `URL` is more appropriate,

File: src/api/route-services/samples.js (hunk 1 above)
Comment (cosa65) [reply src/api/route-services/samples.js:192]: `cellranger_version` is a special header we need to send that, if we follow a certain pattern, is then added as metadata to an s3 file, it is information used by gem2s.

Before, we sent this when uploading the samples in the ui but now we need to send it when generating the url.

**Review by aerlaut — APPROVED** [review]
LGTM

**Pull request merged**